sdks/java/core/src/main/java/org/apache/beam/sdk/coders/AvroCoder.java

```diff
@@ -206,6 +206,16 @@ public Schema get() {
   private final EmptyOnDeserializationThreadLocal<DatumWriter<T>> writer;
   private final EmptyOnDeserializationThreadLocal<DatumReader<T>> reader;
 
+  // Lazily re-instantiated after deserialization
+  @Nullable private transient ReflectData reflectData;
+
+  private ReflectData getReflectData() {
+    if (reflectData == null) {
```

Comment:
Its only idempotent if ReflectData is idempotent and if multiple instantiated copies come to the same result everytime but because of the internal caching ReflectData's may differ.